national_rail/transform.py

```diff
@@ -0,0 +1,153 @@
+""" Module for transforming data received from the National Rail API into a 
+    useful format. """
+
+import logging
+
+from datetime import datetime, timedelta
+import xml.etree.ElementTree as ET
+from bs4 import BeautifulSoup
+
+
+def read_data_from_file(filename: str) -> str:
+    """ Opens file, reads and returns its content as a String. """
+
+    with open(filename, "r", encoding="utf-8") as file:
+        return file.read()
+
+
+def load_tree_root(national_rail_xml: str) -> ET.Element:
+    """ Returns root node of tree (from data.xml) """
+
+    tree = ET.ElementTree(ET.fromstring(national_rail_xml))
+    return tree.getroot()
+
+
+def reverse_tree(root: ET.Element) -> ET.Element:
+    """ Creates a new tree containing the root and its elements in reverse, to allow
+        for traversing the tree in reverse. Returns the new tree by its root. """
+
+    reversed_elements = list(reversed(list(root.iter())))
+    new_root = ET.Element("root")
+
+    for element in reversed_elements:
+        new_root.append(element)
+
+    return new_root
+
+
+def get_incidents(root: str, namespaces: dict) -> ET.Element:
+    """ Extracts each incident from the NationalRail data and returns it. """
+
+    return root.findall('ns:PtIncident', namespaces)
+
+
+def find_text_element(element: ET.Element, path: str, namespaces: str) -> str:
+    """ Searches for element by path and namespace, returns text if element exists. """
+
+    element_found = element.find(path, namespaces)
+    return element_found.text if element_found is not None else None
```

Comment:
```suggestion
    return element_found.text if element_found else None
```

Reply:
I originally had this, but I was given a suggestion to include is not None due to something being deprecated in the future.